index.js

```diff
@@ -95,11 +96,14 @@ Request.prototype._tryUntilFail = function () {
   this.attempts++;
 
   this._req = Request.request(this.options, function (err, response, body) {
+    var delayStrategyResult = this.delayStrategy ? this.delayStrategy(err, response, body) : undefined;
+    var delayStrategyResultValid = typeof delayStrategyResult === "number";
+
     if (response) {
       response.attempts = this.attempts;
     }
     if (this.retryStrategy(err, response, body) && this.maxAttempts > 0) {
-      this._timeout = setTimeout(this._tryUntilFail.bind(this), this.retryDelay);
+      this._timeout = setTimeout(this._tryUntilFail.bind(this), delayStrategyResultValid ? delayStrategyResult : this.retryDelay);
```

Comment:
and replace:

```
delayStrategyResultValid ? delayStrategyResult : this.retryDelay
```

by:

```
this.delayStrategy(err, response, body)
```

indeed, we don't check that delayStrategy yield a number but it should, it's the same logic with retryStrategy that should always yield a boolean so we don't need this extra check here :)